#171648182 add user role settings endpoint

YankyJayChris opened the pull request:
#### Description
It adds an endpoint that enables "Super administrator" users to assign roles to other users.

#### Type of change

Please select the relevant option

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

- [X] Unit
- [ ] Integration
- [ ] End-to-end
- [ ] No testing required

#### How to Test

-clone the repository, run the server locally then follow the instructions from the swagger documentation.

#### Any background context you want to add (Optional)
-N/A

#### STORY ID

[Finishes #171648182]

placideirandora commented on server/controllers/userRoleController.js:
```diff
@@ -0,0 +1,25 @@
+import AuthHelpers from '../helpers/authHelpers';
+
+const updateRole = async (req, res) => {
```
Consider refactoring to use the Class approach as the others.

placideirandora commented on server/controllers/userRoleController.js:
```diff
@@ -0,0 +1,25 @@
+const updateRole = async (req, res) => {
+  const { role } = req.body;
+  const { userId } = req.params;
+  const exists = await AuthHelpers.userExists('id', userId);
```
Consider renaming the variable to **userExists** for meaningfulness.

placideirandora commented on server/controllers/userRoleController.js:
```diff
@@ -0,0 +1,25 @@
+  if (exists) {
+    const userR = await AuthHelpers.updateRole({ userId, role });
+    return res.status(201).json({
+      status: 201,
```
Consider changing the status to **200** as we agreed during the demo.

placideirandora commented on server/helpers/TokenHelper.js:
```diff
@@ -0,0 +1,43 @@
+class TokenHelper {
+
+  /**
+   * Hashs the password.
```
Consider updating the comment. This method does not hash the password, It decodes the token instead.

placideirandora commented on server/helpers/TokenHelper.js:
```diff
@@ -0,0 +1,43 @@
+   * @param {string} secrectKey The secret key.
+   * @returns {string} The users's hashed password.
+   */
+  static decodedToken(token, secrectKey) {
```
Change the function name to **decodeToken** is that's the functionality.

placideirandora commented on server/helpers/TokenHelper.js:
```diff
@@ -0,0 +1,43 @@
+  }
+
+  /**
+   * Hashs the password for signup and login response.
```
Update the comment, according to the method name, it generates a token instead.

placideirandora commented on server/middlewares/tokenValidator.js:
```diff
@@ -0,0 +1,25 @@
+    }
+    return res.status(401).json({ status: 401, error: 'User not recognised. Please create account and try again.' });
+  } catch (error) {
+    return res.status(400).json({ status: 400, error: 'Malformed/ Incorrect security token ! Check token and try again.' });
```
Consider using the same 401 status code here.

placideirandora commented on server/tests/auth.test.js:
```diff
@@ -0,0 +1,45 @@
+            .post('/api/auth/signin')
+            .send(usersTester[1]);
+        res.should.have.status(401);
+        res.body.should.be.an('object');
```
Consider testing the message property as well.

devel605 committed:
171648182-story(user role seting): add user role router

- userhelper file
- isSuperAdmin and roleValidator middleware

[Finishes #171648182]

171648182-story(user role seting): add user role router

-userhelper file
-isSuperAdmin and roleValidator middleware

[Finishes #171648182]

placideirandora committed:
Merge pull request #12 from STACK-UP-3/ft-user-signin-171648169

171648169 Feature of signing in a user

YankyJayChris committed:
171648182-story(user role seting): add user role router

-userhelper file
-isSuperAdmin and roleValidator middleware

[Finishes #171648182]

YankyJayChris committed:
Merge branch 'develop' into ft-user-role-setting-#171648182

YankyJayChris committed:
Merge branch 'develop' into ft-user-role-setting-#171648182

YankyJayChris committed:
Merge branch 'ft-user-role-setting-#171648182' of https://github.com/STACK-UP-3/champs-backend into ft-user-role-setting-#171648182

YankyJayChris committed:
Merge branch 'develop' into ft-user-role-setting-#171648182

YankyJayChris committed:
Merge branch 'develop' into ft-user-role-setting-#171648182
